pkg/ctl/gitops/apply.go

```diff
@@ -0,0 +1,173 @@
+package gitops
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+	"github.com/weaveworks/eksctl/pkg/git"
+	"github.com/weaveworks/eksctl/pkg/gitops"
+	"github.com/weaveworks/eksctl/pkg/gitops/fileprocessor"
+	"github.com/weaveworks/eksctl/pkg/gitops/flux"
+)
+
+type options struct {
+	gitOptions        git.Options
+	quickstartNameArg string
+	outputPath        string
+}
+
+func applyGitops(cmd *cmdutils.Cmd) {
+	cfg := api.NewClusterConfig()
+	cmd.ClusterConfig = cfg
+
+	cmd.SetDescription("apply", "Setting up GitOps and apply a Quick Start profile", "")
+
+	var opts options
+
+	cmd.SetRunFuncWithNameArg(func() error {
+		return doApplyGitops(cmd, opts)
+	})
+
+	cmd.FlagSetGroup.InFlagSet("General", func(fs *pflag.FlagSet) {
+		fs.StringVarP(&opts.quickstartNameArg, "quickstart-profile", "", "", "name or URL of the Quick Start profile. For example, app-dev.")
+		fs.StringVarP(&opts.gitOptions.URL, "git-url", "", "", "URL for the git repository that will contain the cluster components")
+		fs.StringVarP(&opts.gitOptions.Branch, "git-branch", "", "master", "Git branch")
+		fs.StringVarP(&opts.outputPath, "output-path", "", "./", "Path to directory where the GitOps repo will be cloned")
```

Comment:
Also: is this the same as `profile-path` when using `generate profile`?

Reply:
Not exactly :/ I am happy to get name suggestions by the way. This is the directory where the user's repo will be cloned, (if path is "/home/m/myrepos/", it will be cloned in "/home/m/myrepos/repo1/"). Then, in the repo, we will create two folders: `flux` and `base`.